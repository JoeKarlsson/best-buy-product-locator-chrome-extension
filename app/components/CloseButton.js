import React from 'react';
import styled from 'styled-components';
import { COLORS } from '../constants/styles';

const Container = styled.div`
  width: 100%;
  text-align: right;
`;

const X = styled.div`
  width: 100%;
  color: ${COLORS.blue};
  cursor: pointer;
  font-size: 40px;
  font-weight: 700;
`;

const CloseButton = () => (
  <Container onClick={() => window.close()}>
    <X>Ã—</X>
  </Container>
);

export default CloseButton;
