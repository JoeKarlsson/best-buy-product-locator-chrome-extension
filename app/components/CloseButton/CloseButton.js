import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { COLORS } from '../../constants/styles';

const Container = styled.div`
  width: 100%;
  text-align: right;
`;

const X = styled.div`
  width: 100%;
  color: ${COLORS.blue};
  cursor: pointer;
  font-size: 40px;
  font-weight: 700;
`;

const CloseButton = ({ isPopup }) => (
  <Container
    onClick={() => {
      if (isPopup) {
        window.close();
        return;
      }
      document.getElementById('bby-product-locator').style.display = 'none';
    }}
  >
    <X>Ã—</X>
  </Container>
);

CloseButton.propTypes = {
  isPopup: PropTypes.bool.isRequired,
};

export default CloseButton;
