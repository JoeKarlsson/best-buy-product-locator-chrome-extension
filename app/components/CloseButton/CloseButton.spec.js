import 'jsdom-global/register';
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import CloseButton from './CloseButton';

configure({ adapter: new Adapter() });

describe('CloseButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CloseButton />);
  });

  describe('close button onClick', () => {
    it('should call window.close when in popup mode', () => {
      const spy = jest.spyOn(window, 'close');
      wrapper = mount(<CloseButton isPopup />);
      wrapper
        .find('#bby-close')
        .hostNodes()
        .simulate('click');
      expect(spy).toHaveBeenCalled();
    });
    it('should hide the div when not in popup mode', () => {
      document.getElementById = jest.fn(() => ({
        style: { display: 'block' },
      }));
      const spy = jest.spyOn(document, 'getElementById');
      wrapper = mount(<CloseButton isPopup={false} />);
      wrapper
        .find('#bby-close')
        .hostNodes()
        .simulate('click');
      expect(spy).toHaveBeenCalled();
    });
  });

  describe('rendering', () => {
    describe('initial state', () => {
      it('is rendered', () => {
        const component = renderer.create(<CloseButton />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('is rendered correctly', () => {
        expect(wrapper).toHaveLength(1);
      });

      it('should have correct inital instance', () => {
        const initialInstance = wrapper.instance();
        const expectedInstance = null;
        expect(initialInstance).toBe(expectedInstance);
      });
    });
  });
});
