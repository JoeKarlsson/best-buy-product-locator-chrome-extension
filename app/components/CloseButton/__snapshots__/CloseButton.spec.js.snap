// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton rendering initial state is rendered 1`] = `
<div
  className="sc-bdVaJa jZTfEh"
  id="bby-close"
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH ljmoGr"
  >
    Ã—
  </div>
</div>
`;
